optimizer_test.go

```diff
@@ -191,6 +191,7 @@ func TestOptimize(t *testing.T) {
 	f(`a + on(x) group_left(*) (prefix{x="a"})`, `a{x="a"} + on(x) group_left(*) (prefix{x="a"})`)
 	f(`a + on(x) group_right(*) prefix "foo_" b{x="a"}`, `a{x="a"} + on(x) group_right(*) prefix "foo_" b{x="a"}`)
 	f(`a{x="a"} + on(x) group_right(*) prefix "foo_" prefix`, `a{x="a"} + on(x) group_right(*) prefix "foo_" (prefix{x="a"})`)
+	f(`foo{bar="a"} ifnot foo{bar="b"}`, `foo{bar="a"} ifnot foo{bar="a",bar="b"}`)
```

Comment:
```suggestion
	f(`foo{a="a"} ifnot foo{b="b"}`, `foo{a="a"} ifnot foo{a="a",b="b"}`)
```
the reason is that test case should show the situation where optimization makes sense. Having `bar="a",bar="b"` doesn't make sense.